server/server/src/print/label.rs

```diff
@@ -43,6 +44,39 @@ pub async fn print_label_asset(
     }
 }
 
+#[derive(Deserialize)]
+pub struct QueryParams {
+    data: String,
+}
+
+pub async fn get_label_asset(
+    request: HttpRequest,
+    _service_provider: Data<ServiceProvider>,
+    auth_data: Data<AuthData>,
+    query: web::Query<QueryParams>,
+) -> HttpResponse {
+    let auth_result = validate_cookie_auth(request.clone(), &auth_data);
+    match auth_result {
+        Ok(_) => (),
+        Err(error) => {
+            let formatted_error = format!("{:#?}", error);
+            return HttpResponse::Unauthorized().body(formatted_error);
+        }
+    }
+
+    let data: AssetLabelData = match serde_json::from_str(query.into_inner().data.as_str()) {
+        Ok(parsed_data) => parsed_data,
+        Err(err) => {
+            return HttpResponse::BadRequest()
+                .body(format!("Failed to parse data header as JSON: {}", err));
+        }
+    };
+    let zpl = service::print::label::get_asset_label(data);
+    let response = serde_json::json!({ "zpl":zpl });
+
+    HttpResponse::Ok().body(serde_json::to_string_pretty(&response).unwrap())
```

Comment:
There is a way to just send json back, would prefer that, but since we are going to be changing to graphql anyways (see main PR comment) not worth changing now.